File: backend/models/sports.py
from typing import Any
from pydantic import BaseModel


class PlayerStats(BaseModel):
    name: str
    team: str
    position: str
    season: str
    stats: dict[str, Any]


class TeamInfo(BaseModel):
    name: str
    abbreviation: str
    wins: int
    losses: int
    sport: str


class GamePreview(BaseModel):
    home_team: str
    away_team: str
    date: str
    status: str
    home_score: int | None = None
    away_score: int | None = None

File: frontend/src/types/sports.ts
export interface PlayerStats {
  name: string
  team: string
  position: string
  stats: Record<string, number | string>
}

export interface TeamInfo {
  name: string
  abbreviation: string
  wins: number
  losses: number
  sport: 'nba' | 'nfl'
}

export interface GamePreview {
  homeTeam: string
  awayTeam: string
  date: string
  time: string
  sport: 'nba' | 'nfl'
  status: 'scheduled' | 'in_progress' | 'final'
  homeScore?: number
  awayScore?: number
}
